sdk/clientcore/core/src/main/java/io/clientcore/core/utils/IOExceptionCheckedFunction.java

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.core.utils;
+
+import java.io.IOException;
+
+/**
+ * A function that is checked with {@link IOException}.
+ *
+ * @param <T> Input type of the function.
+ * @param <R> Output type of the function.
+ */
+@FunctionalInterface
+public interface IOExceptionCheckedFunction<T, R> {
```

Comment:
Can we extend from `Function<T, R>` here?

```java
public interface CheckedFunction<T, R> extends Function<T, R> {
    @Override
    default R apply(T input) {
        try {
            return applyThrows(input);
        } catch (IOException exception) {
            throw new RuntimeException(exception);
        }
    }

    R applyThrows(T input) throws IOException;
}
```

Reply:
Let's discuss this in another PR given this one is noisy.

I'm not the biggest fan of that as it's a bit clunky.